### ISSUE TITLE
Squiz standard dies during testing on Travis-CI

### ISSUE THREAD

**Issue by photodude** [squizlabs/PHP_CodeSniffer#917]
In trying to figure out why testing with [my custom standard is failing to complete](https://travis-ci.org/photodude/joomla-cms/builds/114939635), I ran a quick [test against the Squiz standard](https://travis-ci.org/photodude/joomla-cms/builds/115124317) in PHPCS 2.x just to make sure things are running right.

The Squiz standard dying on Travis-CI is the same behavior I'm seeing with [my custom standard](https://github.com/photodude/coding-standards/tree/phpcs-2) (which also used part of the Squiz standard). 
- a file that you analyze: Testing against [this Repo](https://github.com/photodude/joomla-cms/tree/phpcs2-testmembervar) [on Travis-CI](https://travis-ci.org/photodude/joomla-cms/builds/115124317)
- standard definition you're using: Squiz (for this test only)
- used PHP_CodeSniffer version: Current stable 2.5.1
- used PHP version: 5.6.5

Note: I'm also seeing this [happening with the MySource standard](https://travis-ci.org/photodude/joomla-cms/builds/115156295)

**Comment by gsherwood** [comment]
I suspect it is this file: administrator/components/com_joomlaupdate/restore.php because it takes a while to run on my machine, but I can't replicate any issues with it getting stuck. My output line for the MySource standard is:

`Processing restore.php [PHP => 54799 tokens in 7652 lines]... DONE in 9.5 secs (19290 errors, 133 warnings)`

Maybe you are silently running out of memory? I would recommend running with `--report=summary` as the number of errors is far too great at the moment. The summary report doesn't record the messages, so uses a lot less memory.

You may also need to bump the memory limit by adding `-d memory_limit=256M` to the command, although I don't run out of memory when I get to restore.php.

If you're not interested in checking JS and CSS standards, then add `--extensions=inc,php` to the command to check far less files in less time.

If would be great if you could provide a command I can run locally to replicate any issues. Try using `-v` on the command line to see what file is being processed when an error occurs.

**Comment by photodude** [comment]
I ran the [test in Travis-CI](https://travis-ci.org/photodude/joomla-cms/builds/115216054#L1656) with the `--report=summary -v -d memory_limit=256M` changes

Travis is killing the test on the `administrator/components/com_joomlaupdate\restore.php` file with the notice `No output has been received in the last 10 minutes`

Seems this file has more tokens, lines, and errors than the other files. but your local test completed in 9.5 seconds where Travis-CI is hanging on the file for more than 10 minutes. Which begs the question of why is Travis-CI taking so long to process the file?

**Comment by gsherwood** [comment]
I'm not sure what PHP version I was using before, but I've now switched to 5.6 and I'm getting an issue with that file. Let me investigate more and I'll get back to you.

**Comment by gsherwood** [comment]
Ok, I figured out why my testing was broken.

This issue has already been fixed in the master branch. Testing with the latest master code (what will become 2.6.0 shortly) works on PHP 5.6

5.5: `Processing restore.php [PHP => 54799 tokens in 7652 lines]... DONE in 8.86 secs (19290 errors, 133 warnings)`
5.6: `Processing restore.php [PHP => 54799 tokens in 7652 lines]... DONE in 6.47 secs (19290 errors, 133 warnings)`
7.0: `Processing restore.php [PHP => 54799 tokens in 7652 lines]... DONE in 3.56 secs (19313 errors, 133 warnings)`

All the same now, just with predictable speed increases.

Can you ignore that file for now? Or use the master branch?

**Comment by photodude** [comment]
It took some work, and a lot of trial and error, but I finally figured out getting the PHPCS master branch and the custom standard [in a state that I can run them on Travis-CI](https://github.com/photodude/joomla-cms/blob/phpcs2-testmembervar/.travis.yml).

[The build hasn't finished running yet](https://travis-ci.org/photodude/joomla-cms/builds/115246548) but it did [process the restore.php](https://travis-ci.org/photodude/joomla-cms/builds/115246548#L1662) and [another file that was hanging](https://travis-ci.org/photodude/joomla-cms/builds/115246548#L3887) when I tried just ignoring the restore.php. I'll take a look at it in the morning to see if it was a total success or not. but so far so good.

Looking forward to the 2.6.0 release

**Comment by gsherwood** [comment]
Looks like it finished and got this result:
```
----------------------------------------------------------------------
A TOTAL OF 399368 ERRORS AND 3697 WARNINGS WERE FOUND IN 5925 FILES
----------------------------------------------------------------------
PHPCBF CAN FIX 356720 OF THESE SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 13 mins, 40.32 secs; Memory: 289.75Mb
```
I don't think I've ever seen 400k errors before. The good news is that the `phpcbf` script can fix 88% of them, if you want to run it. I suspect you don't. Would be one hell of a PR though :)

**Comment by photodude** [comment]
Ya, I'm not surprised by ~400k errors. The main repo for the project is selectively applying the coding standards built on phpcs1.5.x. plus there is a lot of appropriate exclusions for the included vendor libraries.

Now that I can run the draft of the standard with phpcs 2.x on travis-ci we might see much more of this cleaned up. Many of these errors are also coming from vendor libraries which will be excluded when it get's implemented for the live continuous testing. Looks like I have a bug to hunt down in our FileCommentSniff since it's [choking on the Processing](https://travis-ci.org/photodude/joomla-cms/builds/115246548#L8531) of CodeSniffer/Standards/PEAR/Tests/Commenting/FileCommentUnitTest.inc (A file which normally would be an excluded file/folder since it's a vendor item).

I think I have 3 items that still need to be addressed plus this new error and we may finally have [something ready for our community to test](https://github.com/joomla/coding-standards/pull/109) and use.

Thanks for your help and guidance to use `-v` to find the offending file(s).

**Comment by gsherwood** [comment]
> I think I have 3 items that still need to be addressed plus this new error

Is there anything you are waiting on me for, or something you want me to help with?

**Comment by photodude** [comment]
There was the one question I asked on gitter about if there was a possibility that the Squiz.WhiteSpace.MemberVarSpacing could be modified to allow an exception option for the first member var.

And the [New issue in relation to the UselessOverridingMethod I just opened](https://github.com/squizlabs/PHP_CodeSniffer/issues/918)

The other items are just related to figuring out how to get our [standard running phpcs2](https://github.com/photodude/coding-standards/tree/phpcs-2) to match [our written standard](http://joomla.github.io/coding-standards/) and fixing the issue that was just discovered with our custom File Comment Sniff ( [insufficient guarding in the processPackage method](https://github.com/photodude/coding-standards/blob/phpcs-2/Joomla/Sniffs/Commenting/FileCommentSniff.php#L395) )

I'll let you know if we need any specific help or guidance. For now we'll keep plugin along to get things in line.

**Comment by gsherwood** [comment]
Thanks. I'll close this off now as the original issue is resolved.
